test/framework/src/main/java/org/elasticsearch/test/AbstractWireSerializingTestCase.java

```diff
@@ -29,7 +29,7 @@ public abstract class AbstractWireSerializingTestCase<T extends Writeable> exten
      * Copy the {@link Writeable} by round tripping it through {@linkplain StreamInput} and {@linkplain StreamOutput}.
      */
     @Override
-    protected final T copyInstance(T instance, TransportVersion version) throws IOException {
+    protected T copyInstance(T instance, TransportVersion version) throws IOException {
```

Comment:
If this is so deep I'd be worried about making it non-final. I think in my tests I tended to make the `instanceReader` make the PlanStreamInput.

Reply:
Yeah, that's pretty deep...
For the reader I could be good overriding `instanceReader`, but the problem is with the writer, that is [hardcoded in a static method](https://github.com/elastic/elasticsearch/blob/main/test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java#L1777) and I cannot easily override it...

The only solution I see is to change `AbstractWireSerializingTestCase.copyInstance()` to accept a custom writer.